test/e2e/storage/csistoragecapacity.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"context"
+	"time"
+
+	storagev1 "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+
+	"github.com/onsi/ginkgo"
+)
+
+var _ = utils.SIGDescribe("CSIStorageCapacity", func() {
+	f := framework.NewDefaultFramework("csistoragecapacity")
+
+	/*
+		Release: v1.24
+		Testname: CSIStorageCapacity API
+		Description:
+		The storage.k8s.io API group MUST exist in the /apis discovery document.
+		The storage.k8s.io/v1 API group/version MUST exist in the /apis/mode.k8s.io discovery document.
+		The csistoragecapacities resource MUST exist in the /apis/storage.k8s.io/v1 discovery document.
+		The csistoragecapacities resource must support create, get, list, watch, update, patch, delete, and deletecollection.
+	*/
+	framework.ConformanceIt(" should support CSIStorageCapacities API operations", func() {
```

Comment:
Ideally this test is submitted without 'ConformanceIt' for a couple weeks, then when it's shown stable, we add `ConformanceIt()`, updated the `conformance/testdata`. Is that possible?

Reply:
Sure. I've changed this test so that it gets added as a normal test.

https://github.com/kubernetes/kubernetes/pull/108867 has the promotion to conformance.

Reply:
For example, the test runs in `pull-kubernetes-e2e-kind` and should be in any of the periodic storage tests:
```
[It]  should support CSIStorageCapacities API operations
  test/e2e/storage/csistoragecapacity.go:48
[1mSTEP[0m: getting /apis
[1mSTEP[0m: getting /apis/storage.k8s.io
[1mSTEP[0m: getting /apis/storage.k8s.io/v1
[1mSTEP[0m: creating
[1mSTEP[0m: watching
Mar 22 07:53:29.629: INFO: starting watch
[1mSTEP[0m: getting
[1mSTEP[0m: listing in namespace
[1mSTEP[0m: listing across namespaces
[1mSTEP[0m: patching
[1mSTEP[0m: updating
Mar 22 07:53:29.736: INFO: waiting for watch events with expected annotations in namespace
Mar 22 07:53:29.736: INFO: waiting for watch events with expected annotations across namespace
[1mSTEP[0m: deleting
[1mSTEP[0m: deleting a collection
[AfterEach] [sig-storage] CSIStorageCapacity
  test/e2e/framework/framework.go:186
Mar 22 07:53:29.832: INFO: Waiting up to 3m0s for all (but 0) nodes to be ready
[1mSTEP[0m: Destroying namespace "csistoragecapacity-5900" for this suite.

[32m•[0m
[90m------------------------------[0m
{"msg":"PASSED [sig-storage] CSIStorageCapacity  should support CSIStorageCapacities API operations","total":-1,"completed":30,"skipped":286,"failed":0}
```